### PR TITLE
Cleanly exit Chatterino instead of force exiting

### PR DESCRIPTION
**Pull request by Mm2PL**
# Description

Joining the thread stops us from crashing.

Known issues relating to this:
 - if you start and close Chatterino after the recent-messages request was sent but before the response was received, Chatterino will crash. This didn't use to be the case because we'd just throw the process into `_exit`.

### CODE REVIEW

**Comment by pajlada** [comment]
Needs some more testing, specifically on:
 - macOS
 - Windows
 - Windows with browser extension

**Comment by Nerixyz** [comment]
This currently causes a call to std::terminate on Windows, the following fixed it:

```diff
diff --git a/src/providers/liveupdates/BasicPubSubManager.hpp b/src/providers/liveupdates/BasicPubSubManager.hpp
index f82f70363..3a61ee8fe 100644
--- a/src/providers/liveupdates/BasicPubSubManager.hpp
+++ b/src/providers/liveupdates/BasicPubSubManager.hpp
@@ -94,7 +94,10 @@ public:
                 .toStdString());
     }
 
-    virtual ~BasicPubSubManager() = default;
+    virtual ~BasicPubSubManager()
+    {
+        this->stop();
+    }
 
     BasicPubSubManager(const BasicPubSubManager &) = delete;
     BasicPubSubManager(const BasicPubSubManager &&) = delete;
```

We should make sure the app always eventually exits (not like in the tests some time ago, where the Qt event-loop wouldn't spin).

File: src/RunGui.cpp [hunk 1]
```diff
@@ -311,6 +311,7 @@ void runGui(QApplication &a, Paths &paths, Settings &settings)
         pajlada::Settings::SettingManager::gSave();
     }
 
+    thread.join();
```
Comment (pajlada) [inline src/RunGui.cpp:314]:
```suggestion
    if (thread.joinable()) {
        thread.join();
    }
```

File: src/RunGui.cpp [hunk 2]
```diff
@@ -308,13 +311,14 @@ void runGui(QApplication &a, Paths &paths, Settings &settings)
         pajlada::Settings::SettingManager::gSave();
     }
 
+    if (thread.joinable()) {
+        thread.join();
+    }
```
Comment (Mm2PL) [inline src/RunGui.cpp:316]:
```suggestion
    if (thread.joinable())
    {
        thread.join();
    }
```

**Comment by brian6932** [comment]
Tested closing on Windows with the browser extension attached, and had no issues

**Comment by pajlada** [comment]
Latest commit tested on i3 & macOS - works as expected & shuts down pretty fast

**Comment by Nerixyz** [comment]
This was broken by #5058 and #5059 will break this on Windows.

**Comment by pajlada** [comment]
I think I would prefer to do smaller steps towards removing `_exit`, essentially lean into the `fakeDtor` we have now and start cleaning up one part of Chatterino at a time, until fakeDtor essentially runs the full destructor

**Review by pajlada — APPROVED** [review]
image

Works as expected on Linux & macOS - @nerixyz and @brian6932 are testing on Windows

**Comment by pajlada** [comment]
Closing in favour of https://github.com/Chatterino/chatterino2/pull/5537

**Pull request closed without merging**